flaml/autogen/agent/agent.py

```diff
@@ -30,22 +31,79 @@ def _remember(self, memory):
         """Remember something."""
         self._memory.append(memory)
 
-    def _send(self, message, recipient):
+    @staticmethod
+    def _message_to_dict(message: Union[Dict, str]):
+        """Convert a message to a dictionary."""
+        if isinstance(message, str):
+            return {"content": message}
+        else:
+            return message
+
+    def _send(self, message: Union[Dict, str], recipient):
         """Send a message to another agent."""
-        self._conversations[recipient.name].append({"content": message, "role": "assistant"})
+        message = self._message_to_dict(message)
+        # create openai message to be appended to the conversation
+        oai_message = {k: message[k] for k in ("content", "function_call", "name") if k in message}
+        assert oai_message, "Message must contain at least one of the following fields: content, function_call, name."
```

Comment:
I don't think we should assert here. The message is not necessarily an oai message.
@qingyun-wu please advise.

Reply:
The assert prevents the message to be non-oai message.